libcontainer/init_linux.go

```diff
@@ -131,6 +131,25 @@ func finalizeNamespace(config *initConfig) error {
 		return errors.Wrap(err, "close exec fds")
 	}
 
+	// we only do chdir if it's specified
+	doChdir := config.Cwd != ""
+	if doChdir {
+		doChdir = false
+		// First, attempt the chdir before setting up the user.
+		// This could allow us to access a directory that the user running runc can access
+		// but the container user cannot.
+		if err := unix.Chdir(config.Cwd); err != nil {
+			if !os.IsPermission(err) {
```

Comment:
Since this is `unix.`, the errors are bare (unwrapped), and it's _probably_ better to compare directly against `unix.EPERM`.

Reply:
for some reason I thought that wasn't working as expected (though I suspect now i pushed to the wrong branch before...)